### PR TITLE
[SSPROD-27730] Terraform module for s3 onboarding

### PR DESCRIPTION
**Pull request by gi-erre**
# Overall description

This PR introduces the new `cloud-logs` module to enable customer-side provisioning of the necessary resources to enable AWS onboarding.

More specifically: 

* A new `cloud-logs` terraform module has been created under `modules/services`
* The module structure reflects the same one adopted in the other modules
* Small changes to `CONTRIBUTING.md` have been applied in order to reflect the updated process to produce the single `README.md` files that are related to the individual terraform modules

*Note for the reviewers* 
Within the context of this PR, the organizational and `CloudFormation` provisioning cases have not been tackled. Given that, the `cloud-logs` module supports the onboarding of a single AWS account only through `Terraform`. Further developments will need to be carried out in order the other use cases.

### CODE REVIEW

File: modules/services/cloud-logs/variables.tf [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  type        = string
+  description = "(Required) Random string generated unique to a customer"
+}
+
+variable "region" {
+  description = "(Optional) Default region for resource creation"
+  type        = string
+  # This default was inherited from the previous trust-relationship config and related to
+  # organization mode. Does this still hold?
+  default = "eu-central-1"
```
Comment (gi-erre) [inline modules/services/cloud-logs/variables.tf:16]: This default was inherited from the previous trust-relationship config and related to organization mode. Does this still hold?

File: modules/services/cloud-logs/variables.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+variable "external_id" {
+  type        = string
+  description = "(Required) Random string generated unique to a customer"
+}
+
+variable "region" {
+  description = "(Optional) Default region for resource creation"
+  type        = string
+  default     = "eu-central-1"
```
Comment (gi-erre) [inline modules/services/cloud-logs/variables.tf:14]: This default value was inherited from the previous `trust-relationship` config and related to organization mode. Does this still hold, considering that we won't even support organizational accounts right from the start?

File: modules/services/cloud-logs/main.tf [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+# The only resource needed to make cloudingestion start to fetch data from the CloudTrail associated s3 bucket is a
+# properly set AWS IAM Role. Sysdig's trusted identity act as the Principal in the assume role Policy, namely the role
+# that CloudIngestion will use to assume the Client's role. At that point, given the permission set granted to the newly
+# created Role in the Client's account, CloudIngestion will be able to perform all the required actions in order to
+# list and fetch the log files that are automatically published in the target s3 bucket.
+#
+# Note: this setup assumes that the Customer has already properly set up an AWS CloudTrail Trail and the associated bucket.
+# It is responsibility of Sysdig's Secure UI to provide the necessary information to make the Customer perform the
+# necessary operations before applying the Terraform module.
+#-----------------------------------------------------------------------------------------------------------------------
```
Comment (nkraemer-sysdig) [inline modules/services/cloud-logs/main.tf:11]: Please note this repo is public, and as such should not include internal implementation details, even in comments

File: modules/services/agentless-scanning/README.md [hunk 4]
```diff
@@ -6,7 +6,6 @@ The following resources will be created in each instrumented account:
 - An IAM Role and associated policies that allows Sysdig to perform tasks necessary for agentless scanning.
 - A KMS key used to transcript volume snapshots in the each region. Alias for this key in each region.
 
-<!-- BEGINNING OF PRE-COMMIT-TERRAFORM DOCS HOOK -->
```
Comment (nkraemer-sysdig) [inline modules/services/agentless-scanning/README.md:9]: Is the removal of these lines intentional? I believe these are required by pre-commit

File: modules/services/cloud-logs/main.tf (hunk 3 above)
Comment (gi-erre) [reply modules/services/cloud-logs/main.tf:11]: Got it, I'll generalize it a bit!

File: modules/services/agentless-scanning/README.md (hunk 4 above)
Comment (gi-erre) [reply modules/services/agentless-scanning/README.md:9]: As far as I know the pre-commit hook has been deprecated, that's the reason why I removed these lines and updated the contribution guidelines in order to explain how the documentation should be generated. If that's not the case, I can simply remove the single commit where the related changes are now contained

Comment (cgeers) [reply modules/services/agentless-scanning/README.md:9]: > the pre-commit hook has been deprecated

I don't think that's true. There were some broken pre-commit hooks previously and I over-zealously removed all of them. The auto-updates via terraform-docs are valuable, should be re-added, although that does not have been done in this PR.

**Review by cgeers — CHANGES_REQUESTED** [review]
small request to not make changes to other module's readme.

also, aren't we missing a bucket policy here? I had some doubts this was managable independently of the bucket, but when I asked, it was asserted that it was.  Is it not needed? If it is needed, how will it be onboarded? 

otherwise, lgtm.

**Comment by gi-erre** [comment]
> small request to not make changes to other module's readme.

Did it, I removed the related commit, sorry for the misunderstanding :) 

> also, aren't we missing a bucket policy here? I had some doubts this was managable independently of the bucket, but when I asked, it was asserted that it was. Is it not needed? If it is needed, how will it be onboarded?

Turns out that provisioning a IAM `Role` that has `Get` and `List` permissions on the CloudTrail-associated bucket is sufficient for our use case! We already performed a little test with the help of @gmilan-sysdig and @SKosier and everything went smoothly even without the custom bucket policy. That's really good news, since we would have needed to support the append of the policy otherwise, please let me know if you still have doubts on this point!

**Comment by cgeers** [comment]
> Turns out that provisioning a IAM Role that has Get and List permissions on the CloudTrail-associated bucket is sufficient for our use case!

good news indeed.

**Review by cgeers — APPROVED** [review]

**Pull request merged**